lib/frontdesk.rb

```diff
@@ -0,0 +1,117 @@
+require "date"
+
+module Hotel
+  class Frontdesk
+    attr_accessor :rooms, :reservations, :block_reservations
+
+    def initialize
+      @rooms = create_rooms(20)
+      @reservations = []
+      @block_reservations = {}
+    end
+
+    def request_reservation(reservation)
+      if reservation.block_reference == "N/A"
+        assign_room_number(reservation)
+        @reservations << reservation
+        return reservation
+      else
+        @reservations.each do |res|
+          if res.block_reference.eql?(reservation.block_reference) && res.block_availability.eql?(:AVAILABLE)
+            if res.reserved_nights != reservation.reserved_nights
+              raise ArgumentError, "Reservations for #{res.block_reference} must be booked from #{res.checkin_date} to #{res.checkout_date}."
+            else
+              res.instance_variable_set(:@block_availability, :UNAVAILABLE)
```

Comment:
If you want to give the `FrontDesk` thew ability to directly set the block availability instance variable, provide an accessor. This works, but it's doing an end-run around the private instance variables.